3 - FightClub/Code/Objects/EnergyDrink.cs

```diff
@@ -0,0 +1,25 @@
+﻿using ObjectOrientedProblems.Code.Enums;
+using ObjectOrientedProblems.Code.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Text; 
+
+namespace FightClub.Code.Objects
+{
+    class EnergyDrink : IPotion
+    {
+        public int Healing { get; set; }
+        public PotionType Type { get; set; }
+        public EnergyDrink()
+        {
+            Healing = 2;
+            Type = PotionType.Basic;
+        }
+        public void SetType(PotionType potionType)
+        {
+            Type = potionType;
+            if (potionType == PotionType.Greater) { Healing = 5; }
+            else { Healing = 2; }
```

Comment:
Though this code does work, it has potential to fail in the future. It is a hard coded assumption of "is greater" or "other". What if a value gets added to the enum? In situations like enums its preferred to use a switch statement with "default" being an exception raised for "other".

Secondly, if you want to try to "shrink" your if statements (which it looks like you are trying to do) you can use the short hand if statement, known as a `ternary operator`. It would look like this:

```
Healing = (potionType == PotionType.Greater) ? 5 : 2;
```